### ISSUE TITLE
SchedulerPredicates e2e tests running for an hour each in the master upgrade jobs

### ISSUE THREAD

**Issue by skriss** [kubernetes/kubernetes#43228]
Example:

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-1.5-1.6-upgrade-master/21

SchedulerPredicates tests, which are serial, are waiting for all test namespaces to be deleted before executing. The current time limit for waiting is an hour. In the upgrade-master jobs, this is never succeeding resulting in each of these taking an hour to run (doing nothing).  It's not immediately clear from the logs which testing namespace is causing the issue.

@krousey @ixdy @ethernetdan

**Comment by calebamiles** [comment]
Could someone from @kubernetes/sig-scheduling-bugs please help with triage of this test? Thanks!

**Comment by ConnorDoyle** [comment]
For context, here's the framework util function that waits on NS cleanup: https://github.com/kubernetes/kubernetes/blob/0afdcfcaf6c7828276e24cb4e0f15191c749fc7f/test/e2e/framework/util.go#L890

**Comment by timothysc** [comment]
So these appear to be 1.5 e2es running on an upgraded 1.6 cluster.  *affinity* will fail due to api-promotion.

**Comment by ethernetdan** [comment]
@wojtek-t any ideas on why this is timing out?

**Comment by timothysc** [comment]
We know why, there was a long conversation on slack today.. There are 2 parts.  

1. Lingering namespace from another test 
2. Upgrade test using 1.5 test suite on a 1.6 cluster is doomed to fail do to incompatible changes.  

So issues here are not related to the actual test from what we could tell.

**Comment by skriss** [comment]
Digging into this a bit more it appears that the orphaned namespace causing some of these to wait for an hr is actually coming from a prior SchedulerPredicate test. See this:

I0315 19:53:13.163] STEP: Destroying namespace "e2e-tests-sched-pred-c45xc" for this suite.
I0315 19:58:13.899] Mar 15 19:58:13.898: INFO: namespace: e2e-tests-sched-pred-c45xc, resource: bindings, ignored listing per whitelist
I0315 19:58:14.050] Mar 15 19:58:14.049: INFO: namespace: e2e-tests-sched-pred-c45xc, resource: pods, items remaining: 1
I0315 19:58:14.191] Mar 15 19:58:14.191: INFO: namespace: e2e-tests-sched-pred-c45xc, DeletionTimetamp: 2017-03-15 19:53:13 -0700 PDT, Finalizers: [kubernetes], Phase: Terminating
I0315 19:58:14.214] Mar 15 19:58:14.214: INFO: namespace: e2e-tests-sched-pred-c45xc, total namespaces: 4, active: 3, terminating: 1
I0315 19:58:14.216] Mar 15 19:58:14.216: INFO: POD            NODE                                          PHASE    GRACE  CONDITIONS
I0315 19:58:14.217] Mar 15 19:58:14.216: INFO: without-label  gke-bootstrap-e2e-default-pool-b087d004-d053  Pending  30s    [{PodScheduled True 0001-01-01 00:00:00 +0000 UTC 2017-03-15 19:53:12 -0700 PDT  }]
I0315 19:58:14.217] Mar 15 19:58:14.216: INFO: 
I0315 19:58:14.217] Mar 15 19:58:14.216: INFO: Couldn't delete ns: "e2e-tests-sched-pred-c45xc": namespace e2e-tests-sched-pred-c45xc was not deleted with limit: timed out waiting for the condition, pods remaining: 1, pods missing deletion timestamp: 0 (&errors.errorString{s:"namespace e2e-tests-sched-pred-c45xc was not deleted with limit: timed out waiting for the condition, pods remaining: 1, pods missing deletion timestamp: 0"})

(https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-gci-1.5-container_vm-1.6-upgrade-master/38?log#log)

3 tests before the first 1-hr timeout.

**Comment by timothysc** [comment]
@skriss those failed deletes were on 1.5 e2es from the same cluster before the upgrade test?

**Comment by skriss** [comment]
@timothysc 1.5 e2e's, on the same cluster, after the master upgrade. the upgrade test jobs run the E2E's in serial, this namespace delete error happened ~3 tests before the first one-hour hang (waiting for namespace deletion).

**Comment by ethernetdan** [comment]
@skriss @timothysc do we want to add a cleanup step after the 1.5 e2es or is there a more elegant way to fix this?

**Comment by skriss** [comment]
@wojtek-t do you have a sense of whether the 1-hr timeout here is still necessary?

https://github.com/kubernetes/kubernetes/commit/6b6b4098148e1afa289526f785a1a578b4d157be

As a quick fix for this issue just lowering the timeout value to something less than an hr would help.

**Comment by ethernetdan** [comment]
These tests are significantly blurring signal, can we disable of the initial run of these tests and only run on an upgraded cluster?

I'd like this to be fixed by this weekend so we can have a better signal to release.

**Comment by ethernetdan** [comment]
@ixdy how hard would this be to do?

**Comment by ixdy** [comment]
I really don't understand how kubetest runs the upgrade tests. cc @fejta @krousey

**Comment by krousey** [comment]
@ixdy I think adding a --ginkgo.skip=something to the GINKGO_TEST_ARGS env in the job description would do it.

**Comment by skriss** [comment]
I think we can just skip all SchedulerPredicates tests in the upgrade-cluster jobs (via ^^^) given that they're almost all expected to fail for one reason or another anyway.

**Comment by ixdy** [comment]
I'm a little worried about skipping the SchedulerPredicates tests - if they really are consistently failing and providing no signal, we should just delete them.

**Comment by skriss** [comment]
They're problematic specifically for when running 1.5 tests against a 1.6 cluster, largely because of the breaking changes for Affinity/Taints. So not useless in general, just in this particular test configuration.

**Comment by davidopp** [comment]
Hang on a minute. :)

There is nothing wrong with the scheduler predicate tests.

Some of them don't work against a 1.6 cluster. I can give you a list of which ones are expected to fail, if you want. But it is a minority.

**Comment by davidopp** [comment]
And to be clear, "some of them don't work against a 1.6 cluster" means some of the 1.5 release branch scheduler tests don't work against a 1.6 cluster. The 1.6 release branch scheduler tests do work against a 1.6 cluster, of course. And the version skew problem should not happen in the future.

**Comment by ethernetdan** [comment]
@davidopp any ideas for speeding up these runs / avoiding the namespace deletion issue?

**Comment by timothysc** [comment]
There are two distinct problems that exist: 

1. Timeout on namespace cleanup.  This is likely an issue with the core framework b/c the scheduler_predicate tests don't do anything special. 
2. Executing 1.5 tests against 1.6 - this is known to fail.

**Comment by davidopp** [comment]
I checked
https://k8s-testgrid.appspot.com/release-1.6-upgrade-skew#gce-1.5-1.6-upgrade-master
and
https://k8s-testgrid.appspot.com/release-1.6-upgrade-skew#gce-1.5-1.6-upgrade-cluster
and the scheduler predicate tests that are failing are the ones we'd expect to fail, as @timothysc alluded to above. So I think probably we can close this issue (especially since #43394 has been closed), unless someone else thinks otherwise.

**Comment by ethernetdan** [comment]
@davidopp the issue here is not the failures but the amount of time spent trying to delete namespaces, it is causing several upgrade jobs to timeout. I'm hesitant to release with jobs timing out.

**Comment by ncdc** [comment]
@davidopp I'm not 100% sure about this, but it maybe looks like the apiserver is allowing in a pod that subsequently fails validation in the kubelet, and maybe the kubelet isn't able to delete said pod. What I'm seeing is there's a pod that is scheduled to a node, it's listed as Pending, and it's never deleted, which is why the namespace never gets deleted.

With these upgrade tests, is such a scenario possible (apiserver OKs pod, kubelet rejects it)?

cc @derekwaynecarr

**Comment by ncdc** [comment]
@aveshagarwal do you have any idea? I'm seeing:

```
I0315 14:00:19.378] • Failure [360.923 seconds]
I0315 14:00:19.379] [k8s.io] SchedulerPredicates [Serial]
I0315 14:00:19.379] /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:826
I0315 14:00:19.379]   validates that a pod with an invalid podAffinity is rejected because of the LabelSelectorRequirement is invalid [It]
I0315 14:00:19.379]   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:434
I0315 14:00:19.379] 
I0315 14:00:19.379]   Mar 15 13:55:18.645: Expect error of invalid, got : <nil>
I0315 14:00:19.379] 
I0315 14:00:19.379]   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:429
```

and

```
I0315 13:55:19.183] STEP: Destroying namespace "e2e-tests-sched-pred-sv87h" for this suite.
I0315 14:00:19.302] Mar 15 14:00:19.301: INFO: namespace: e2e-tests-sched-pred-sv87h, resource: bindings, ignored listing per whitelist
I0315 14:00:19.349] Mar 15 14:00:19.349: INFO: namespace: e2e-tests-sched-pred-sv87h, resource: pods, items remaining: 1
I0315 14:00:19.370] Mar 15 14:00:19.370: INFO: namespace: e2e-tests-sched-pred-sv87h, DeletionTimetamp: 2017-03-15 13:55:19.185552465 -0700 PDT, Finalizers: [kubernetes], Phase: Terminating
I0315 14:00:19.373] Mar 15 14:00:19.373: INFO: namespace: e2e-tests-sched-pred-sv87h, total namespaces: 4, active: 3, terminating: 1
I0315 14:00:19.377] Mar 15 14:00:19.377: INFO: POD                                                 NODE                             PHASE    GRACE  CONDITIONS
I0315 14:00:19.377] Mar 15 14:00:19.377: INFO: without-label-b281be85-09c1-11e7-9c16-0242ac110002  bootstrap-e2e-minion-group-spz7  Pending  30s    [{PodScheduled True 0001-01-01 00:00:00 +0000 UTC 2017-03-15 13:55:18.648131096 -0700 PDT  }]
I0315 14:00:19.377] Mar 15 14:00:19.377: INFO: 
I0315 14:00:19.378] Mar 15 14:00:19.377: INFO: Couldn't delete ns: "e2e-tests-sched-pred-sv87h": namespace e2e-tests-sched-pred-sv87h was not deleted with limit: timed out waiting for the condition, pods remaining: 1, pods missing deletion timestamp: 0 (&errors.errorString{s:"namespace e2e-tests-sched-pred-sv87h was not deleted with limit: timed out waiting for the condition, pods remaining: 1, pods missing deletion timestamp: 0"})
```

and (kubelet.log):

```
W0316 00:55:59.373570   21381 config.go:343] Pod[4] (without-label-b281be85-09c1-11e7-9c16-0242ac110002.e2e-tests-sched-pred-sv87h) from api failed validation, ignoring: metadata.annotations.scheduler.alpha.kubernetes.io/affinity.podAffinity.requiredDuringSchedulingIgno│
redDuringExecution[0].topologyKey: Required value: can only be empty for PreferredDuringScheduling pod anti affinity
```

**Comment by ncdc** [comment]
It looks like the kubelet filters out pods that fail validation: https://github.com/kubernetes/kubernetes/blob/fbc320af28a11c09d487f9b7cea43fe8f8b224c6/pkg/kubelet/config/config.go#L231, called on line 278. I'm assuming that this is why a pod that makes it into the apiserver but fails validation in the kubelet can't be deleted gracefully.

**Comment by ncdc** [comment]
I have to run for now. cc @yujuhong for additional sig-node coverage. I would love it if someone could explain how a pod passes apiserver validation but fails kubelet. I'm guessing it's because the apiserver and kubelet are running different validation code (given this is an upgrade test)?

**Comment by derekwaynecarr** [comment]
@ncdc -- your analysis appears correct to me.

**Comment by derekwaynecarr** [comment]
@ncdc -- there are potential remedies.  for example, if the kubelet fails any pod that fails local validation rather than leave pending namespace deletion will work as expected.  we cannot have the namespace controller force delete, so that may be the preferred option if we think this is must fix.

**Comment by vishh** [comment]
+1 on having kubelet fail pods that do not pass local validation.

**Comment by derekwaynecarr** [comment]
@vishh @yujuhong -- i am not able to work this issue the remainder of the day.  the challenge is that where this validation check is done i cannot figure out the cleanest way to handle the failure.  most times where we fail things (active deadline for example), its driven by the sync loop, but in this case, the sync loop won't even see the pod.  i guess we could pass something that knows how to do permafail of a pod into the `NewPodConfig` , and when running in standalone mode, we would just do a no-op?  needs more thought than i can give it the rest of today.

**Comment by yujuhong** [comment]
The validation code also guards the potentially invalid pod manifest file (e.g., they may even have invalid names?). One potential solution is to move the validation code down to the main kubelet pod admission logic and reject it right there. Not sure if kubelet would run into any issue if the manifest file is very broken. It might be ok.

That said, this is not a trivial change and it's so close to the release. My gut reaction is to avoid fixes like this. Can we have someone confirm whether the pod failing the validation is expected (as @ncdc suggested in https://github.com/kubernetes/kubernetes/issues/43228#issuecomment-288226125), and whether this is something we must fix?

**Comment by ethernetdan** [comment]
@yujuhong who would be a good person to decide on that?

Also if we decide this doesn't need to be fixed in the kubelet, can we just cleanup the Pod inside the test logic?

**Comment by yujuhong** [comment]
@dchen1107 is looking into this. She'll probably update this with more information soon.

**Comment by dchen1107** [comment]
I haven't looked into the issue deeply, but if what @ncdc's comment at https://github.com/kubernetes/kubernetes/issues/43228#issuecomment-288222469 is the root cause of the issue, this might be a serious release blocker here with API object validation logic.

If an API object is validated by 1.6 release, it should be validated by 1.5 release too. The validation policy shouldn't be weaken over the releases based on Kubernetes upgrade / deprecated / backward compatibility policy.

cc/ @caesarxuchao @lavalamp

**Comment by dchen1107** [comment]
There are two possible components might causing this very issue here:

1) API conversion logic upon convert versioned object to API object in 1.6, breaks the backward compatibility here
2) Validation code in 1.6 might have a weaken rule. 

Still looking into the detail.

**Comment by lavalamp** [comment]
Can we have the offending pod so we can see what exactly it is?

Kubelet should be rejecting (failing) pods that don't validate, not ignoring them.

**Comment by lavalamp** [comment]
This appears to have something to do with the scheduler annotations.

**Comment by lavalamp** [comment]
(reading more) ah, we expect these tests to fail, the issue is that the pod never gets cleaned up.

**Comment by lavalamp** [comment]
The issue is that the bug is in a 1.5 kubelet, I guess.

**Comment by caesarxuchao** [comment]
https://github.com/kubernetes/kubernetes/blob/release-1.5/pkg/api/validation/validation.go#L2089

This function is removed in 1.6, so the annotation was not validated in the 1.6 apiserver, but 1.5 kubelet still ran the check and failed.

**Comment by lavalamp** [comment]
Yeah, that stuff got moved to fields in 1.6.

**Comment by davidopp** [comment]
Thanks for the debugging, everyone.

I didn't even realize kubelet runs validation, I thought it just ran [predicates](https://github.com/kubernetes/kubernetes/blob/0bc16d8966c38c8b3e2af57f9addce254031b247/pkg/kubelet/lifecycle/predicate.go#L52).

So is the solution to put back the validation we removed?

**Comment by lavalamp** [comment]
Dawn has a good point, that if kubelet failed these pods, we'd have a chance at producing a recidivism type problem where the same pod keeps getting scheduled over and over.

So, yes, the solution is to put back the validation.

**Comment by lavalamp** [comment]
Arguably we *do* need a solution for graceful deletion when kubelet is ignoring a pod.

**Comment by lavalamp** [comment]
But I think that's a separate issue.

**Comment by dchen1107** [comment]
Yes, the solution sgtm. Again, Kubelet shouldn't fail pod due to mismatching validations / API backward compatibility. This is different from the admission control. 

@lavalamp But I agreed that Kubelet shouldn't silent ignore a pod. I am filing a separate issue for that. Thanks everyone.

**Comment by yujuhong** [comment]
> Dawn has a good point, that if kubelet failed these pods, we'd have a chance at producing a recidivism type problem where the same pod keeps getting scheduled over and over.

That doesn't mean kubelet should not surface this information through an event, etc, but see my comment above: https://github.com/kubernetes/kubernetes/issues/43228#issuecomment-288241585. The code is shared by pods from all sources, so it is run as early as possible to avoid kubelet operating on an invalid pod. We *could* move this down to the admission logic if we are sure this won't cause any issue, or we *could* consider distinguishing pod sources (which is always a pain).

None of the above can fix the 1.5 kubelet though, so restoring the validation logic is still the only "solution".

**Comment by aveshagarwal** [comment]
But wonder why the kubelet is throwing the following error:

```
redDuringExecution[0].topologyKey: Required value: can only be empty for PreferredDuringScheduling pod anti affinity
```

when the test `validates that a pod with an invalid podAffinity is rejected because of the LabelSelectorRequirement is invalid` does have topology key, unless I am missing something?

**Comment by dchen1107** [comment]
@davidopp I am assigning this to you to address or delegate the backward incompatibility issue here.

**Comment by davidopp** [comment]
@aveshagarwal Do you think you could add back the annotation validation logic we removed in these PRs?
https://github.com/kubernetes/kubernetes/pull/37299
https://github.com/kubernetes/kubernetes/pull/39478
https://github.com/kubernetes/kubernetes/pull/38957
https://github.com/kubernetes/kubernetes/pull/38871

Note: I didn't check if #41617 already took care of the first two -- those might already be done.

**Comment by aveshagarwal** [comment]
@davidopp do we want to add validation check back for taints/tolerations too given that taints.tolerations annotations are completely removed in 1.6?
 
Also adding back validation check can fix only those tests where apiserver is not throwing errors but kubelet is. Are we sure there are no others upgrade tests related to pod/node affinity and taints/tolerations that might be failing due to some other reasons? 

Also I think https://github.com/kubernetes/kubernetes/pull/41617 does not add validation logic back for affinity related tests, thats why tests are failing.

**Comment by derekwaynecarr** [comment]
I think we need a solution kubelet side in future to not leave pods in limbo.  If I understand this issue correctly, it's never safe to validate alpha fields like this as dropping the alpha feature in a future release would break old kubelets.   That just seems wrong in the long run.  For example, would we be able to drop this logic in 1.7 for the old alpha annotation form?

**Comment by derekwaynecarr** [comment]
Could the node controller force delete these pods as well that are stuck in kubelet limbo?

**Comment by davidopp** [comment]
@aveshagarwal 

> do we want to add validation check back for taints/tolerations too given that taints.tolerations annotations are completely removed in 1.6?

Yes I think we have to, as people might create taint/toleration annotation with 1.6 master that is then consumed by 1.5 kubelet where it fails validation. Even though taint/toleration annotation is meaningless in 1.6, user might create it anyway.

> Are we sure there are no others upgrade tests related to pod/node affinity and taints/tolerations that might be failing due to some other reasons?

There are kubectl skew tests that failed due to moving annotation to fields but we disabled the combination that is expected to fail. Also some of the other upgrade test cases failed but we've been making sure only the ones that are expected to fail due to annotation->field change are failing.

**Comment by derekwaynecarr** [comment]
@davidopp @lavalamp @dchen1107 - i understand the desire is to maintain the legacy annotation validation code in 1.6 to appease 1.5 kubelets, but i am trying to reconcile that near term desire with the long term direction pitched here:

https://docs.google.com/document/d/1YpYMLZiFfGHB4KNNdS-JY7uX8w5H9tzyq7-lWrFTmN8/edit#heading=h.3sjjx2gvn9hz

it sounds like there are some assumptions related to kubelet skew that are worth calling out.

>alpha field will stop actuating anything 
this is only true if you don't land on n-2 kubelets

>alpha field will continue to exist in the API spec for one release, but it will serve a purely informational role
it needs to exist in API spec for n-2 release?

>system will treat the alpha field like it treats annotations today
i still wonder if we should ever do validation of alpha stuff given this rule, especially if kubelet validates alpha features.

**Comment by ethernetdan** [comment]
It's looking like unless a solution can be implemented by tomorrow morning we will likely have to block the release for this issue.

/cc @kubernetes/release-maintainers

**Comment by derekwaynecarr** [comment]
I do not understand why we would block the release.

1. alpha features on GA resources are not guaranteed to work across releases
1. alpha features on GA resources that are removed in 1.6 masters, but cause validation failures in 1.5 clients seems like a never win situation.  it implies we support migration of alpha features, which it seems like as a project, we said we would not support per @lavalamp proposal (see https://docs.google.com/document/d/1YpYMLZiFfGHB4KNNdS-JY7uX8w5H9tzyq7-lWrFTmN8/edit?disco=AAAAA8X2pnc)

It seems like at best, we should document that users that consumed these alpha features may need to force delete those pods when their nodes are skewed to their master.  Or we need to say any field validation of alpha resources needs to persist for N-2 versions (which I think is a mistake).

What am I missing?

**Comment by ethernetdan** [comment]
Was thinking of blocking based on Slack conversation with @dchen1107

@dchen1107, what are your thoughts?

**Comment by smarterclayton** [comment]
I don't see his as being a supported configuration.  Alpha features being dropped... means alpha features are dropped.

**Comment by ethernetdan** [comment]
That logic makes sense to me

**Comment by ethernetdan** [comment]
We should modify the tests to compensate for this failure and fix our upgrade signal.

**Comment by aveshagarwal** [comment]
So is my understanding correct that now we are not going to add annotation based validation for pod/node affinity and taints/tolerations back?

**Comment by davidopp** [comment]
@aveshagarwal Please start working on this. We can discuss later.

**Comment by calebamiles** [comment]
@davidopp are you referring to the tests in https://github.com/kubernetes/kubernetes/issues/43228#issuecomment-288282437?

**Comment by davidopp** [comment]
I'd like Avesh to implement adding back the validation logic that I described in my earlier comment.

**Comment by derekwaynecarr** [comment]
Ack.  I think the policy described previously should not get retroactively applied in these cases.  It does mean we need to carry this validation until 1.8 unless I am mistaken given kubelet skew.

**Comment by derekwaynecarr** [comment]
I would like us to tackle how kubelet should enforce validation for alpha fields in 1.7+ to avoid a recurrence of this problem.  Validation does not appear feature gate flagged.

**Comment by ethernetdan** [comment]
I see the number one priority for the release as stopping these from taking an hour each to run.

**Comment by davidopp** [comment]
The thing @aveshagarwal is working on will fix that.

**Comment by aveshagarwal** [comment]
I have added validation logic back in https://github.com/kubernetes/kubernetes/pull/43498 .

**Comment by dchen1107** [comment]
I didn't realize that this is an alpha feature, and agreed that this shouldn't be the release blocker for 1.6.

**Comment by davidopp** [comment]
It is and is not an alpha feature. Pod and node affinity/anti-affinity are alpha features we moved to beta in 1.6 but a lot of people are using the alpha form so we wanted to make it possible for people to upgrade to 1.6 without breaking their scheduling, even though it is alpha. (Unfortunately because it moves from annotation to field, it's not possible to "prepare" for the upgrade, since the field doesn't exist in 1.5.)

Anyway, @aveshagarwal has a PR out to fix it already, and we need that to make the tests stop running for an hour and then timing out anyway, which his PR will do (in addition to fixing the user-facing aspect) so it's correct for this to be a release blocker. If all goes well, it will be merged by the time people wake up tomorrow.

**Comment by wojtek-t** [comment]
So the fix was merged 4 hours ago. However, the current run of upgrade tests:
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-1.5-1.6-upgrade-master/32/
still doesn't contain it (as it was started 1 hour before that). So we will have to wait for the next one.

**Comment by skriss** [comment]
@ixdy is it possible to restart any running upgrade jobs that started before the fix was merged so we don't have to wait so long?

**Comment by ethernetdan** [comment]
@ixdy +1 if that's possible

On Mar 22, 2017 8:08 AM, "skriss" <notifications@github.com> wrote:

@ixdy <https://github.com/ixdy> is it possible to restart any running
upgrade jobs that started before the fix was merged so we don't have to
wait so long?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/kubernetes/issues/43228#issuecomment-288428706>,
or mute the thread

**Comment by ethernetdan** [comment]
Merged, looking for fix in upgrade results. Will reopen if issue persists.

**Comment by davidopp** [comment]
@ixdy killed the builds that started before 3 pm today, so we should hopefully have something useful in the next 12 hours.
